src/main/java/com/example/auctionapp/services/ItemService.java

```diff
@@ -31,23 +31,44 @@
 
-    public Page<Item> getAllItems(int page, int size, String order, String orderColumn, Long superCategoryId) {
+    public Page<Item> getAllItems(int page, int size, String order, String orderColumn, Long superCategoryId, Long[] categories, int[] prices) {
         PageRequest pageable;
-        if (superCategoryId != null) {
-            pageable = PageRequest.of(page, size, Sort.by(orderColumn).ascending());
-            return itemRepository.getBySupercategory(superCategoryId, pageable);
-        } else {
-            if (order != null && orderColumn != null) {
-                if (order.equals("asc")) {
-                    pageable = PageRequest.of(page, size, Sort.by(orderColumn).ascending());
+        Page<Item> statePage;
+        if (prices != null) {
+            if (categories != null) {
+                statePage = itemRepository.findByCategoryCategoryIdInAndStartingPriceBetween(Arrays.asList(categories), prices[0], prices[1], PageRequest.of(page, size, Sort.by(orderColumn).ascending()));
+            } else if (superCategoryId != null) {
+                statePage = itemRepository.findByCategorySupercategoryIdAndStartingPriceBetween(superCategoryId, prices[0], prices[1], PageRequest.of(page, size, Sort.by(orderColumn).ascending()));
+            } else {
+                if (order != null && orderColumn != null) {
+                    if (order.equals("asc")) {
+                        pageable = PageRequest.of(page, size, Sort.by(orderColumn).ascending());
+                    } else {
+                        pageable = PageRequest.of(page, size, Sort.by(orderColumn).descending());
+                    }
                 } else {
-                    pageable = PageRequest.of(page, size, Sort.by(orderColumn).descending());
+                    pageable = PageRequest.of(page, size);
                 }
+                statePage = itemRepository.findByStartingPriceBetween(prices[0], prices[1], pageable);
+            }
+        } else {
+            if (categories != null) {
+                statePage = itemRepository.findByCategoryCategoryIdIn(Arrays.asList(categories), PageRequest.of(page, size, Sort.by(orderColumn).ascending()));
+            } else if (superCategoryId != null) {
+                statePage = itemRepository.findByCategorySupercategoryId(superCategoryId, PageRequest.of(page, size, Sort.by(orderColumn).ascending()));
             } else {
-                pageable = PageRequest.of(page, size);
+                if (order != null && orderColumn != null) {
+                    if (order.equals("asc")) {
+                        pageable = PageRequest.of(page, size, Sort.by(orderColumn).ascending());
+                    } else {
+                        pageable = PageRequest.of(page, size, Sort.by(orderColumn).descending());
+                    }
+                } else {
+                    pageable = PageRequest.of(page, size);
+                }
+                statePage = itemRepository.findAll(pageable);
```

Comment:
this is really hard to read, and from perspective of debugging will be nightmare. Also it would be hard to add new filter options. It is nice thing to use predefined JPA `findBy` methods, but in this case maybe it would be better to write sql query with all filters. And provide them all here. If some of them are null, it should be ok, and should not affect results.